pkg/client/client.go

```diff
@@ -215,11 +275,37 @@ func NewValidatedClient(ctrlRuntimeClient client.Client, secretName, namespace,
 	}
 
 	// Check that the endpoint can be resolved by the endpoint resolver.
+	// If the endpoint is not resolvable locally, we try to validate using the AWS API.
```

Comment:
There's quite a lot of logic being nested here now, perhaps we should extract this into a `getEndpoint` function to cover all of this?